src/praisonai-agents/praisonaiagents/permissions/command_parser.py

```
@@ -0,0 +1,260 @@
+
+    try:
+        tokens = shlex.split(segment, comments=False, posix=True)
+    except ValueError:
+        # Unbalanced quotes etc. — conservative fallback: whole segment.
+        tokens = segment.split()
+
+    op = ShellOp()
+    args: List[str] = []
+    skip_next = False
+
+    for idx, tok in enumerate(tokens):
```

Ruff (0.15.18)

[warning] Loop control variable `idx` not used within loop body

Rename unused `idx` to `_idx`

(B007)